pkg/remote/remote.go

```diff
@@ -25,19 +28,46 @@ import (
 var RemoteAddressRegex = regexp.MustCompile(`(?i)^((ssh|sftp)://)+(?P<credentials>[^:@]+(:([^:@])+)?@)?[^:]+(:\d+)?`)
 
 const (
-	defaultPort        = 22
 	defaultTimeoutSecs = 20
 	// Max file size supporting downloading from remote. Default is 512Kb.
 	maxFileSize = 512000
 )
 
+var defaultKeyFiles = []string{ // nolint:gochecknoglobals
+	"~/.ssh/id_dsa",
+	"~/.ssh/id_ecdsa",
+	"~/.ssh/id_ecdsa_sk",
+	"~/.ssh/id_ed25519",
+	"~/.ssh/id_ed25519_sk",
+	"~/.ssh/id_rsa",
+}
+
 // Client communicates using sftp protocol.
 type Client struct {
-	User string
-	Pass string
-	Host string
-	Port int
-	Path string
+	User            string
+	Pass            string
+	HostKeyAlias    string
+	HostDisplayName string
+	HostDerived     string
+	Port            int
+	Path            string
+}
+
+type HostSet map[string]struct{}
+
+func (i HostSet) Contains(hosts []string) bool {
```

Comment:
It's a simple iteration over an array of strings and don't need a method for it especially public. Method is when a struct is needed to call any func:
```Go
func (x struct) MyFunc(args..) bool {
    ....
}
```

I'd suggest:
```Go
func contains(values []string, args ...string) bool {
	for _, val := range values {
		for _, arg := range args {
			if val == arg {
				return true
			}
		}
	}

	return false
}
```

Reply:
I'd also move it to bottom since it has less importance then `WithDetection`, `NewClient`, etc...